curl/src/client.cc

```diff
@@ -173,6 +171,65 @@ namespace leatherman { namespace curl {
         return res;
     }
 
+    void client::download_file(std::string const& url, std::string const& file_path, std::string const& token)
+    {
+        // Reset the options
+        curl_easy_reset(_handle);
+
+        // Set common options
+        curl_easy_setopt_maybe(url, file_path, CURLOPT_NOPROGRESS, 1);
+        curl_easy_setopt_maybe(url, file_path, CURLOPT_URL, url.c_str());
+
+        CURLcode result = CURLE_OK;
+        char errbuf[CURL_ERROR_SIZE] = { '\0' };
+        try {
+            auto temp_path = fs::path(file_path).parent_path() / fs::unique_path("temp_file_%%%%-%%%%-%%%%-%%%%");
+            auto fp = boost::nowide::fopen(temp_path.string().c_str(), "wb");
+            if (!fp) {
+                throw http_file_download_exception(url, file_path, _("Failed to open temporary file for writing"));
+            }
+
+            curl_easy_setopt_maybe(url, file_path, CURLOPT_WRITEFUNCTION, write_file);
+            curl_easy_setopt_maybe(url, file_path, CURLOPT_WRITEDATA, fp);
+
+            // TODO: How long should connection time be? Is 5 minutes too much?
+            curl_easy_setopt_maybe(url, file_path, CURLOPT_CONNECTTIMEOUT, 300);
+            set_ca_info(url, file_path);
+            set_client_info(url, file_path);
+
+            // Authentication header
+            curl_list x_auth_header;
+            x_auth_header.append("X-Authentication: " + token);
```

Comment:
This shouldn't be necessary. It can also be handled externally with add_headers.